proposals/0000-publish-arbitrary-data.md

```diff
@@ -0,0 +1,67 @@
+---
+Title: Publish arbitrary data instruction
+Number: 0
+Status: Draft
+Version: 4
+Authors:
+ - Francisco Aguirre
+Created: 2023-11-16
+Impact: Low
+Requires:
+Replaces:
+---
+
+## Summary
+
+This RFC proposes a new instruction, `Publish`, to publish some data to another location.
+
+## Motivation
+
+XCM itself is not a good means of storing and retrieving arbitrary data.
+This fixes the storage part, leaving the retrieval to another mechanism.
+
+## Specification
+
+The new instruction looks like this:
+
+```rust
+Publish { data: BoundedVec<Key, Value, Bound> }
+```
+
+`Key` and `Value` are arrays of bits, they could be anything.
+
+The `Bound` value should be configurable.
+
+### Examples
+
+A program like the following might be sent to another location for publishing some data:
+
+```rust
+WithdrawAsset(/* ... */),
+BuyExecution { /* ... */ }
+Publish { data: [(b"My key", b"My value")] }
```

Comment:
What is the origin of this data? Can that be stored along with the data itself? Or is this operation by default restricted to a specific origin? E.g., I am a parachain interested in this data but only if it comes from another parachain, not from users within it.

Reply:
If there's some additional logic as to who can send/execute this instruction, it should be added here.